Fw/Cmd/CmdString.hpp

```diff
@@ -1,37 +1,59 @@
-#ifndef FW_CMD_STRING_TYPE_HPP
-#define FW_CMD_STRING_TYPE_HPP
+// ======================================================================
+// @file   CmdString.hpp
+// @author F Prime
+// @brief  A string sized for a command argument
+// ======================================================================
+
+#ifndef FW_CMD_STRING_HPP
+#define FW_CMD_STRING_HPP
 
 #include <FpConfig.hpp>
-#include <Fw/Types/StringType.hpp>
-#include <Fw/Cfg/SerIds.hpp>
+
+#include "Fw/Cfg/SerIds.hpp"
+#include "Fw/Types/StringBase.hpp"
 
 namespace Fw {
 
-    class CmdStringArg : public Fw::StringBase {
-        public:
+class CmdStringArg final : public StringBase {
```

Comment:
Is this PR done?  I expected to review a bunch of changes to string types where they implement from `ExternalString` instead it seems that the code was just moved from the `.cpp` files to the headers....without inheriting implementation from external string

Reply:
Yes, it's done. I decided not to have the implementations inherit from ExternalString. I realized nothing was gained by doing that, because in C++ operator= is not inherited. Instead I removed the C++ operator= implementations that were copy-pasting code from the superclass operator= and replaced them with direct calls to StringBase::operator=. The resulting code is much simpler and less error-prone and is functionally identical to what we had before -- in particular there's no overhead from inheriting from ExternalString. If this isn't clear, let's discuss offline.